scripts/gen_json/gen_json.py

```diff
@@ -0,0 +1,393 @@
+        self._stdout = sys.stdout
+        sys.__stdout__ = self
+        sys.stdout = self
+
+    def write(self, data):
+        pass
+
+    def __getattr__(self, item):
+        if item in self.__dict__:
+            return self.__dict__[item]
+
+        return getattr(self._stdout, item)
+
+    def reset(self):
+        sys.stdout = self._stdout
+
+
+def run(output_path, lvgl_config_path, output_to_stdout, *compiler_args):
+    # stdout = STDOut()
+
+    # The thread is to provide an indication that things are being processed.
+    # There are long periods where nothing gets output to the screen and this
+    # is to let the user know that it is still working.
+    if not output_to_stdout:
+        event = threading.Event()
+
+        def _do():
+            while not event.is_set():
+                event.wait(1)
+                sys.stdout.write('.')
+                sys.stdout.flush()
+
+            print()
+
+        t = threading.Thread(target=_do)
+        t.daemon = True
+        t.start()
+
+    temp_directory = tempfile.mkdtemp(suffix='.lvgl_json')
+    lvgl_path = project_path
+    lvgl_src_path = os.path.join(lvgl_path, 'src')
+    lvgl_header_path = os.path.join(lvgl_path, 'lvgl.h')
+    temp_lvgl = os.path.join(temp_directory, 'lvgl')
+
+    try:
+        os.mkdir(temp_lvgl)
+        shutil.copytree(lvgl_src_path, os.path.join(temp_lvgl, 'src'))
+        shutil.copyfile(lvgl_header_path, os.path.join(temp_lvgl, 'lvgl.h'))
+
+        lvgl_header_path = os.path.join(temp_lvgl, 'lvgl.h')
+        pp_file = os.path.join(temp_directory, 'lvgl.pp')
+
+        if lvgl_config_path is None:
+            lvgl_config_path = os.path.join(lvgl_path, 'lv_conf_template.h')
+
+            with open(lvgl_config_path, 'rb') as f:
+                data = f.read().decode('utf-8').split('\n')
+
+            for i, line in enumerate(data):
+                if line.startswith('#if 0'):
+                    data[i] = '#if 1'
+                else:
+                    for item in (
+                        'LV_USE_LOG',
+                        'LV_USE_OBJ_ID',
+                        'LV_USE_OBJ_ID_BUILTIN',
+                        'LV_USE_FLOAT',
+                        'LV_USE_BIDI',
+                        'LV_USE_LODEPNG',
+                        'LV_USE_LIBPNG',
+                        'LV_USE_BMP',
+                        'LV_USE_TJPGD',
+                        'LV_USE_LIBJPEG_TURBO',
+                        'LV_USE_GIF',
+                        'LV_BIN_DECODER_RAM_LOAD',
+                        'LV_USE_RLE',
+                        'LV_USE_QRCODE',
+                        'LV_USE_BARCODE',
+                        'LV_USE_TINY_TTF',
+                        'LV_USE_GRIDNAV',
+                        'LV_USE_FRAGMENT',
+                        'LV_USE_IMGFONT',
+                        'LV_USE_SNAPSHOT',
+                        'LV_USE_FREETYPE'
+                    ):
+                        if line.startswith(f'#define {item}'):
```

Comment:
Unrelated

```suggestion
                        if line.startswith(f'#define {item} '):
```

I'm getting `#define LV_USE_OBJ_ID_BUILTIN   0` becoming `#define LV_USE_OBJ_ID 1`

Reply:
check again.

```json
{
            "name": "LV_USE_OBJ_ID",
            "json_type": "macro",
            "docstring": "",
            "params": null,
            "initializer": "0 "
        },
        {
            "name": "LV_USE_OBJ_ID_BUILTIN",
            "json_type": "macro",
            "docstring": "",
            "params": null,
            "initializer": "0 "
        },
```

There are 2 macros one is `LV_USE_OBJ_ID` and the other is `LV_USE_OBJ_ID_BUILTIN`. If a config file is not given using the `--lvgl-config="path/to/lv_conf.h"` command line parameter the script will use the template to generate a config file.

Reply:
I see what you are saying.